packages/core/src/execution-engine/node-execution-context/webhook-context.ts

```diff
@@ -172,4 +174,16 @@ export class WebhookContext extends NodeExecutionContext implements IWebhookFunc
 			itemIndex,
 		);
 	}
+
+	validateExecutionWaitingToken() {
+		const req = this.getRequestObject();
+		const token = req.query[WAITING_TOKEN_QUERY_PARAM];
+
+		if (typeof token !== 'string') return false;
+
+		const expectedToken = this.getExecutionWaitingToken();
+
+		const valid = crypto.timingSafeEqual(Buffer.from(token), Buffer.from(expectedToken));
```

Comment:
The comment from cubic is valid. Don’t ignore it

Reply:
yes, had to wrapped in trycatch block, thought I did this, thanks